src/Form/RedirectForm.php

```diff
@@ -133,7 +117,7 @@ public function validate(array $form, FormStateInterface $form_state) {
     $parsed_url = UrlHelper::parse(trim($source['path']));
     $path = isset($parsed_url['path']) ? $parsed_url['path'] : NULL;
     $query = isset($parsed_url['query']) ? $parsed_url['query'] : NULL;
-    $hash = Redirect::generateHash($path, $query, $form_state->getValue('language'));
+    $hash = Redirect::generateHash($path, $query, $form_state->getValue('language')[0]['value']);
```

Comment:
Oh, I see. The parent::validate() call actually returns a built entity. Can we maybe do $redirect = parent::validate() and then get the langcode (and other things) from there? Right now this ties us to the used widget, and it might be hidden or so...